src/mpi/comm/commutil.c

```diff
@@ -1110,6 +1110,22 @@ int MPIR_Comm_delete_internal(MPIR_Comm * comm_ptr)
      * communicator must not be freed.  That is the reason for the
      * test on mpi_errno here. */
     if (mpi_errno == MPI_SUCCESS) {
+        /* receive any unmatched messages to clear the queue. this could be the result of user error,
+         * or send operations that were unable to be canceled by the device layer */
+        MPIR_Object_add_ref(comm_ptr);
+        int flag;
+        MPI_Status status;
+        do {
+            mpi_errno = PMPI_Iprobe(MPI_ANY_SOURCE, MPI_ANY_TAG, comm_ptr->handle, &flag, &status);
```

Comment:
We are inside global lock, need exit the lock to prevent recursive locking.

Reply:
I think it is better to use `MPID_Iprobe`.

Reply:
fixed